src/tests/ftest/util/dmg_utils.py

```diff
@@ -23,44 +23,183 @@
-        self.mode = FormattedParameter("--mode={}")
-        self.size = FormattedParameter("--size={}")
-        self.nvme = FormattedParameter("--nvme={}")
-        self.svcn = FormattedParameter("--svcn={}")
-        self.target = FormattedParameter("--target={}")
-        self.force = FormattedParameter("--force", False)
-        self.pool = FormattedParameter("--pool={}")
-        self.svc = FormattedParameter("--svc={}")
-        self.rank = FormattedParameter("--rank={}")
-        self.cont = FormattedParameter("--cont={}")
-        self.oid = FormattedParameter("--oid={}")
-
-    def get_param_names(self):
-        """Get a sorted list of dmg command parameter names."""
-        names = self.get_attribute_names(FormattedParameter)
-        names.extend(["request", "action"])
-        return names
+        self.insecure = FormattedParameter("-i", True)
+        self.debug = FormattedParameter("-d", False)
+        self.json = FormattedParameter("-j", False)
+
+    def _get_action_command(self):
+        """Assign a command object for the specified request and action."""
+        if self.action.value == "format":
+            self.action_command = self.DmgFormatSubCommand()
+        elif self.action.value == "prepare":
+            self.action_command = self.DmgPrepareSubCommand()
+
+    class DmgFormatSubCommand(CommandWithParameters):
+        """Defines an object representing a format sub dmg command."""
+
+        def __init__(self):
+            """Create a dmg Command object."""
+            super(DmgCommand.DmgFormatSubCommand, self).__init__(
+                "/run/dmg/format/*", "format")
+            self.force = FormattedParameter("-f", False)
+
+    class DmgPrepareSubCommand(CommandWithParameters):
+        """Defines a object representing a prepare sub dmg command."""
+
+        def __init__(self):
+            """Create a dmg Command object."""
+            super(DmgCommand.DmgPrepareSubCommand, self).__init__(
+                "/run/dmg/prepare/*", "prepare")
+            self.pci_wl = FormattedParameter("-w {}")
+            self.hugepages = FormattedParameter("-p {}")
+            self.targetuser = FormattedParameter("-u {}")
+            self.nvmeonly = FormattedParameter("-n", False)
+            self.scmonly = FormattedParameter("-s", False)
+            self.force = FormattedParameter("-f", False)
+            self.reset = FormattedParameter("--reset", False)
+
+
+def storage_scan(hosts):
+    """ Execute scan command through dmg tool to servers provided.
+
+    Args:
+        hosts (list): list of servers to run scan on.
+
+    Returns:
+        Avocado CmdResult object that contains exit status, stdout information.
+
+    """
+    # Create and setup the command
+    dmg = DmgCommand(get_file_path("bin/daos_shell"))
+    dmg.request.value = "storage"
+    dmg.action.value = "scan"
+    dmg.insecure.value = True
+    dmg.hostlist.value = hosts
+
+    try:
+        result = dmg.run()
```

Comment:
`ExecutableCommand.run()` currently does not return anything - some additional changes needed in `command_utils.py` to achieve this - the tricky part is what to return if `self.run_as_process` is True: `None`? and empty `CmdResult` object?

Reply:
Returning None in the case should be fine I think. If we at some point try to get the value from `var = ExecutableCommand.run()` wouldn't `var = None` if it was run as a subprocess? 
So if we did:
```
if self.run_as_subprocess:
     self._run_subprocess()
else:
     return self._run_process()
```
So we'd just have to check if `var = ExecutableCommand.run()` is not None and check if it's an instance of CmdResult object.

Reply:
Well the method should always return something.  So it could be:
```
def run(self):
    if self.run_as_subprocess:
        self._run_subprocess()
        return None
    return self._run_process()
```

Reply:
From your suggestion I'll change this to:
```
        if self.run_as_subprocess:
            self._run_subprocess()
            return None
        return self._run_process()
```